test/q-repeat.test.js

```diff
@@ -14,3 +15,56 @@ describe('map', () => {
     });
   });
 });
+
+describe('retry', () => {
+  it('repeats until success', () => {
+    let delay = 500;
+    let invocations = 0;
+
+    return Q.timeoutRetry(150, () => {
+      delay -= 100;
+      invocations += 1;
+      return Q(`answer ${invocations}`).delay(delay);
+    }).then((result) => {
+      expect(delay).to.eq(100);
+      expect(invocations).to.eq(4);
+      expect(result).to.eq('answer 4');
+    });
+  });
+
+  it('fails fast if actual error occurs', () => {
+    return Q.timeoutRetry(500, () => {
+      throw new Error('omg');
+    }).catch((err) => {
+      expect(err.message).to.eq('omg');
+    });
+  });
+
+  it('hits outer timeout', () => {
```

Comment:
The topic of canceling promises brings out some very strong opinions on both sides.  Since the q camp is on the side of never-cancel, I've added support for the accepted construct for canceling a promise, which is to pass in a token that the task itself populates with a cancel method.  It's not really useful for where I plan to use these (see above), but it's there if it's needed.

http://stackoverflow.com/a/30235261/793449